pkg/tfgen/docs.go

```diff
@@ -899,10 +899,14 @@ func (p *tfMarkdownParser) parseImports(subsection []string) {
 			continue
 		}
 
+		// Remove the shell comment characters to avoid writing this line as a Markdown H1:
+		section = strings.TrimPrefix(section, "# ")
```

Comment:
I'm wondering if it isn't confusing to have comments rendered as normal text, since they often are written in a different style and with a different purpose. We could render them as comments instead, e.g., by wrapping them in backticks?

Reply:
tfgen's code is fairly fragile and difficult to test. Practically speaking, you can eyeball the generate schemas of a provider or two, but you can't really view how the docs will look in the registry.

And because of the large number of providers and the unstructured nature of docs, I'm taking a super-duper incremental approach here: Just make the Snowflake issue better without (hopefully) making any of the other providers worse.

(I did consider your approach while working on this fix and decided it wasn't worth the risk of a potential regression.)

Reply:
fair enough!